collections/_article/introducing-reverse-z.md

```diff
@@ -0,0 +1,135 @@
+
+```
+POSITION = vec4(VERTEX.xy, 1.0, 1.0);
+```
+
+And that's it!
+
+Since this code is used so widely, we are adding a special [warning](https://github.com/godotengine/godot/pull/90587) for it so the engine can give you a heads up if your code will potentially break in 4.3.
+
+Importantly, writes to ``POSITION`` will only break if they are writing values in directly in clip space. If you are transforming values from view space using the ``PROJECTION_MATRIX``, no code changes are necessary. For example, the following code will continue to work:
+
+```
+POSITION = PROJECTION_MATRIX * MODELVIEW_MATRIX * vec4(VERTEX, 1.0);
+```
+
+### Writes to ``DEPTH``
+
+Writing to ``DEPTH`` comes with the same warnings as writing to ``POSITION``. If the clip space value comes from transforming a view space value with the ``PROJECTION_MATRIX``, then no changes are necessary. 
+
+```
+// This will continue to work
+vec4 clip_pos = PROJECTION_MATRIX * vec4(VERTEX, 1.0);
+clip_space.xyz /= clip_space.w;
+DEPTH = clip_space.z;
+
+// This will need to change
+DEPTH = 0.0; // Needs to change to 1.0
+```
+
+### Reads from depth_texture
+
+This is potentially the area where the most changes will be necessary. Like the above two cases, as long as you are doing your operations in another space (e.g. view space), you don't need to worry. The following very common code ([from the documentation](https://docs.godotengine.org/en/4.2/tutorials/shaders/screen-reading_shaders.html#depth-texture)) will continue to work as before:
+
+```
+float depth = textureLod(depth_texture, SCREEN_UV, 0.0).r;
+vec4 upos = INV_PROJECTION_MATRIX * vec4(SCREEN_UV * 2.0 - 1.0, depth, 1.0);
+vec3 pixel_position = upos.xyz / upos.w;
+```
+
+For example, in the above code, I may be comparing the depth buffer value to my pixel's depth value. So I could do something like draw a foam outline in a water shader. i.e.:
+
+```
+if (VERTEX.z > pixel_position.z) {
+    // Do something.
+}
+```
+
+This code will require no modification. Where users will need to make modifications in their code is if they are doing these operations in clip space. For example, instead of transforming the depth into view space and comparing the position, users could instead transform the vertex position into clip space and compare. 
+
+```
+vec4 clip_pos = PROJECTION_MATRIX * vec4(VERTEX, 1.0);
+clip_space.xyz /= clip_space.w;
+float depth = textureLod(depth_texture, SCREEN_UV, 0.0).r;
+
+if (clip_pos > depth) {
+    // Do something.
+}
+```
+
+This case will need to be modified. Particularly, the conditional needs to be flipped to become:
+
+```
+if (clip_pos < depth) {
+    // Do something.
+}
+```
+
+This reflects the fact that the near plane is now at a clip space z position of 1.0 instead of 0.0. other similar cases will arise. For example, you may instead be looking at the difference between the depth buffer and your vertex position:
```

Comment:
```suggestion
This reflects the fact that the near plane is now at a clip space z position of 1.0 instead of 0.0. Other similar cases will arise. For example, you may instead be looking at the difference between the depth buffer and your vertex position:
```